tianshou/data/utils.py

```diff
@@ -18,7 +18,11 @@ def to_numpy(x: Union[
     elif isinstance(x, Batch):
         x.to_numpy()
     elif isinstance(x, (list, tuple)):
-        x = [to_numpy(x_i) for x_i in x]
+        x = np.asanyarray(x)
+        if x.dtype == np.object:
+            x = [to_numpy(e) for e in x]
+        else:
+            x = to_numpy(x)
```

Comment:
> List of object are converted individually, as a single tensor otherwise.

Why is that? You say you want to support tuple observation, and it needs ``x = [to_numpy(x_i) for x_i in x]``.

Reply:
The following code fails immediately:
```python
a = np.asanyarray([torch.ones((3, 4)), torch.ones((3, 4))])
```